src/rate_limiter.py

```
@@ -0,0 +1,21 @@
+from collections import defaultdict
+from datetime import datetime, timedelta
+from flask import abort
+
+# Simple rate limiter using memory (use Redis for production)
+rate_limit = defaultdict(list)
+
+def rate_limit(max_requests=100, window_minutes=1):
```

Ruff (0.12.2)

Redefinition of unused `rate_limit` from line 6

(F811)